### DISCUSSION TITLE
How to bypass the URL check when adding the Navigation Node pnp.core - C#

### DISCUSSION THREAD

**Discussion by fischkoder** [pnp/pnpcore#1330] [Q&A]
Hi all,

I am currently working on Adding Navigations(TopNavigationBar, QuickLaunch) of SharePoint Site by using PnP.Core C#.

I try to use 
`context.Web.Navigation.QuickLaunch.AddAsync(new NavigationNodeOptions
  {
      Title = displayName,
      Url = itemUrl,
      ParentNode = parentNode
  );`
to add new Navigation node to the root site.
But I found that this API will remove the root domain in the Url then only apply the rest part of Url to Navigation node, which will cause 500 error 'no such file or folder'
For example, if I pass the itemUrl = "https://contoso.sharepoint.com/sites/testsite",  the response will show me "/sites/testsite " instead.
I know there is a way to set the Navigation node url as External, but in PnP.Core library, IsExternal it's a read only property now.

So how can I configure the complete/full Url to the navigation node without any removal of root domain?

**Comment by jansenbe** [comment]
@fischkoder : PnP Core SDK is not making the URL relative, it's SharePoint that does that. Can you try with specifying a more complete URL (e.g. https://contoso.sharepoint.com/sites/testsite/SitePages/Home.aspx) to see if that makes a difference?

**Reply by fischkoder** [reply]
Thanks for your advice, and I solved it by giving the navigation node a pure 'non-internal' link such as 'https://linkless.header' (http will make this node as a Label), then in the SharePoint side, this node's will be initialized as a 'External' link permanently. then give the actual link to this node. I've tried the complete URL with 'https://', it still performed the same error.
